server/src/main/java/org/elasticsearch/index/mapper/BinaryFieldMapper.java

```diff
@@ -63,26 +61,34 @@
         }
     }
 
-    public static class Builder extends FieldMapper.Builder<Builder> {
+    private static BinaryFieldMapper toType(FieldMapper in) {
+        return (BinaryFieldMapper) in;
+    }
+
+    public static class Builder extends ParametrizedFieldMapper.Builder {
+
+        final Parameter<Boolean> stored = Parameter.boolParam("store", false, m -> toType(m).stored, false);
+        final Parameter<Boolean> hasDocValues = Parameter.boolParam("doc_values", false, m -> toType(m).hasDocValues,  false);
 
         public Builder(String name) {
-            super(name, Defaults.FIELD_TYPE);
-            hasDocValues = false;
-            builder = this;
+            super(name);
+        }
+
+        // For testing
```

Comment:
If you make them package private that does a pretty good job implying they are only exposed for testing.

Reply:
It's used outside the package, annoyingly.  I've changed it so that if you need to set it programmatically you can use a constructor parameter.